src/visualization/associations/a_priori_canvas.py

```diff
@@ -0,0 +1,236 @@
+from PyQt5.QtWidgets import QWidget, QVBoxLayout
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+import plotly.graph_objects as go
+
+
+class APrioriScatterPlot(FigureCanvasQTAgg):
+    HIGHLIGHTED_COLOR = "#f17300"
+    REGULAR_COLOR = "#F5F5F5"
+    HIGHLIGHTED_RULE_A_COLOR = "#81a4cd"
+    HIGHLIGHTED_RULE_B_COLOR = "#054a91"
+    HIGHLIGHTED_RULE_A_B_COLOR = "#f17300"
+
+    def __init__(self, fig: plt.Figure, axes: plt.Axes, transaction_sets):
+        self.annot = None
+        self.sc = None
+        self.axes = axes
+        fig.tight_layout()
+        self.transaction_sets = transaction_sets
+        self._assign_random_positions()
+        self.fig = fig
+
+        super().__init__(fig)
+
+    def plot_set(self, highlighted_set: tuple = None):
+        self.axes.cla()
+        self.axes.axis("off")
+
+        self.annot = self.axes.annotate(
+            "",
+            xy=(0, 0),
+            xytext=(-20, 0),
+            textcoords="offset points",
+            bbox=dict(boxstyle="round", fc="w"),
+        )
+        self.annot.set_visible(True)
+
+        mask = self._get_points_mask_set(set(highlighted_set) if highlighted_set is not None else None)
+        self.sc = self.axes.scatter(
+            list(compress(self.x_positions, mask)), list(compress(self.y_positions, mask)),
+            color=self.HIGHLIGHTED_COLOR, s=10, zorder=2, label="containing the frequent set"
+        )
+
+        inverse_mask = list(map(operator.not_, mask))
+        self.axes.scatter(
+            list(compress(self.x_positions, inverse_mask)), list(compress(self.y_positions, inverse_mask)),
+            color=self.REGULAR_COLOR, s=10, zorder=1, label="not containing"
+        )
+
+        self.axes.legend(fontsize=5)
+        self.fig.canvas.mpl_connect("motion_notify_event", self.hover)
+        self.draw()
+
+    def plot_rule(self, set_a: tuple, set_b: tuple):
+        self.axes.cla()
+        self.axes.axis("off")
+
+        mask_a, mask_b, mask_a_b, mask_remaining = self._get_points_mask_rule(set(set_a), set(set_b))
+        self.axes.scatter(
+            list(compress(self.x_positions, mask_a)), list(compress(self.y_positions, mask_a)),
+            color=self.HIGHLIGHTED_RULE_A_COLOR, s=10, zorder=3, label="A",
+        )
+
+        self.axes.scatter(
+            list(compress(self.x_positions, mask_b)), list(compress(self.y_positions, mask_b)),
+            color=self.HIGHLIGHTED_RULE_B_COLOR, s=10, zorder=2, label="B",
+        )
+
+        self.axes.scatter(
+            list(compress(self.x_positions, mask_a_b)), list(compress(self.y_positions, mask_a_b)),
+            color=self.HIGHLIGHTED_RULE_A_B_COLOR, s=10, zorder=4, label="A and B",
+        )
+
+        self.axes.scatter(
+            list(compress(self.x_positions, mask_remaining)), list(compress(self.y_positions, mask_remaining)),
+            color=self.REGULAR_COLOR, s=10, zorder=1, label="~(A or B)",
+        )
+
+        self.axes.legend(fontsize=5)
```

Comment:
Maybe increase fontsize a little. I see that is difficult to put everything in small section so I left decision for you.